### DISCUSSION TITLE
Vcenter tags

### DISCUSSION THREAD

**Discussion by ShmoeTheHo** [bb-Ricardo/netbox-sync#64] [Ideas]
Any way we get this to import vcenter tags? Thats my missing piece for ansible :)

**Comment by bb-Ricardo** [comment]
Hi,

I assume this should be possible. I have to look into it.

So the script should add all vCenter tags to NetBox as well?

**Reply by ShmoeTheHo** [reply]
Yup, and apply as tags or some sort of meta in netbox as well. That would be awesome :)

**Comment by bb-Ricardo** [comment]
There is also an issue open for that: #65 

The Problem is that tags are not part of the regular API. I would need to add another connection to query tags for all objects.

**Comment by bb-Ricardo** [comment]
Hey @ShmoeTheHo can you try this `feature/read-vcenter-tags`. I added syncing of vCenter tags.

you need to set `sync_tags` and `sync_parent_tags` to `True` in your `settings.ini`

would highly appreciate your test results.

**Comment by maw1146** [comment]
Hey @bb-Ricardo,
I tried to sync tags from vCenter adding the sync_tags and sync_parent_tags to the settings.ini.
However when I try to run the sync, I have the following warning:
"WARNING: Unable to import Python 'vsphere-automation-sdk'. Tag syncing will be disabled."
and the tag is not synced.

Could you please help me out? :)

**Reply by bb-Ricardo** [reply]
of course.

you need to install the  'vsphere-automation-sdk': Install instructions here: https://github.com/bb-Ricardo/netbox-sync#vmware-tag-sync-if-necessary

let me know if it works

**Reply by bb-Ricardo** [reply]
you might need to use `pip3` as command, depending on your OS.

**Reply by maw1146** [reply]
Hi,
actually I could have readed the doc, sorry :) 

however, I've installed the vsphere-automation following the 2 commands. but now when I try to run the netbox-sync.py I have a new issue (that before I never saw).
Here is the output:

Traceback (most recent call last):
  File "./netbox-sync.py", line 22, in <module>
    from module.netbox.connection import NetBoxHandler
  File "/opt/netbox-sync/module/netbox/connection.py", line 17, in <module>
    import urllib3
  File "/usr/lib/python3.6/site-packages/urllib3/__init__.py", line 8, in <module>
    from .connectionpool import (
  File "/usr/lib/python3.6/site-packages/urllib3/connectionpool.py", line 11, in <module>
    from .exceptions import (
  File "/usr/lib/python3.6/site-packages/urllib3/exceptions.py", line 2, in <module>
    from .packages.six.moves.http_client import (
ModuleNotFoundError: No module named 'urllib3.packages.six'



based on pip, urllib3 and six are both correctly installed.

**Reply by bb-Ricardo** [reply]
you have to make sure your virtual environment must still be active. did you do a?
```
. .venv/bin/activate
```

**Reply by maw1146** [reply]
it's working, thanks :)

**Comment by ShmoeTheHo** [comment]
I just saw this myself and will be testing it out.

On Sat, Oct 16, 2021 at 5:25 AM Ricardo ***@***.***> wrote:

> Hey @ShmoeTheHo <https://github.com/ShmoeTheHo> can you try this
> feature/read-vcenter-tags. I added syncing of vCenter tags.
>
> you need to set sync_tags and sync_parent_tags to True in your
> settings.ini
>
> would highly appreciate your test results.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bb-Ricardo/netbox-sync/discussions/64#discussioncomment-1486308>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFZRCBPZLKOZ53GIU332KSDUHFAHPANCNFSM44ZUZDBA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Sent from Gmail Mobile
